### ISSUE TITLE
Terraform Cannot Find Darwin Package for Provider Version 0.2.11

### ISSUE THREAD

**Issue by mphammer** [harness/terraform-provider-harness#151]
### Terraform Version
```
$ terraform version
Terraform v1.2.2
on darwin_amd64
```

### Affected Resource(s)
- terraform init

### Terraform Configuration Files
```hcl
terraform {
  required_version = ">= 0.14.8"
  required_providers {
    harness = {
      source  = "harness/harness"
      version = "0.2.10"
    }
  }

provider "harness" {
  endpoint   = "https://app.harness.io/gateway"
  account_id = <account_id>
  api_key    = <api_key>
}
```

### Debug Output
https://gist.github.com/mphammer/7de76df595b802c1d937af94aae63285

### Expected Behavior
`terraform init` finds the darwin_amd64 package for provider version 0.2.10 and initializes.

### Actual Behavior
`terraform init` fails to initialize because it can't find a package for darwin_amd64 even though I can see it on the GitHub releases page.  
It should also not be complaining about 0.2.11 because the versions are all set to 0.2.10.

### Steps to Reproduce
1. Setup a terraform repo on a macbook, specify provider version 0.2.10, and run `terraform init`.

**Comment by micahlmartin** [comment]
Hi @mphammer . It looks like you have a bad reference somewhere in your state. I see this:
```
registry.terraform.io/harness-io/harness v0.2.11
```

But it should be this
```
registry.terraform.io/harness/harness v0.2.11
```

This was change quite a few versions back where we move the repo from `harness-io/terraform-provider-harness` to `harness/terraform-provider-harness`

Let me know if that makes sense.

**Comment by mphammer** [comment]
Thank you for the clarification! This fixed the problem. 

For anyone else that might find this: 

Step 1: I updated the source to be harness/harness and I updated the version
```
  required_providers {
    harness = {
      source  = "harness/harness"
      version = "0.2.11"
    }
  }
```
Step 2: I pulled the statefile from remote, changed all instances of `harness-io` to `harness`, and then pushed back to remote.
Step 3: I ran a `terraform init` and it passed.
